Introduces better VaRA documentation (#636)
se-passau/VaRA-Tool-Suite: docs/source/research_tool_docs/vara/vara.rst

****
VaRA
****

> VaRA is an analysis framework that enables users to build static and dynamic analyses for analyzing high-level concepts, using advanced compiler and analysis technology in the background.

Reviewer: ```suggestion
VaRA is an analysis framework that enables users to build static and dynamic analyses for analyzing high-level concepts using advanced compiler and analysis technology in the background.
```

> Our goal is to enable the user to build these analyses by only focusing on the highlevel conceptual information that should be analyzed, without worrying about low-level details, such as, building complicated compiler modifications or configuring precise but difficult to use static analyses.

Reviewer: ```suggestion
Our goal is to enable the user to build these analyses by only focusing on the high-level conceptual information that should be analyzed without worrying about low-level details, such as building complicated compiler modifications or configuring precise but difficult to use static analyses.
```

The figure below, gives a rough overview of VaRA and the analysis process.

> In general, VaRA and our modified clang compiler take in source code together with highlevel conceptual information and either directly analyze it, using various static analyses, or generate an instrumened binary, which can run different dynamic analyses.

Reviewer: ```suggestion
In general, VaRA and our modified clang compiler take as input source code together with high-level conceptual information and either directly analyze it, using various static analyses, or generate an instrumented binary, which can run different dynamic analyses.
```
